### DISCUSSION TITLE
How to read/write values of complex properties?

### DISCUSSION THREAD

**Discussion by obscuresc** [ChristianTremblay/BAC0#445] [General]
Hi, 

I'm developing a HVAC product and am using BAC0 to test its BACnet interface.
I want to write tests against properties for R/W access, or in some cases, 'ignored' writes where a platform limitation does not allow control of R/W permissions. 

How do I go about writing to object identifier, status flags, event state for example?

Thanks!

EDIT: Looking for complex properties, like arrays of status flags, event state etc.

**Comment by bbartling** [comment]
maybe look at some of these examples this was code I used for interacting on a live DDC system. Use Python 3.11 or below. Py 3.12 still needs some updates for the BAC0 repo.

https://gist.github.com/bbartling/88e95557b0b297bea721c0f285657d53

**Reply by obscuresc** [reply]
Sorry, I can't see an example beyond r/w presentValue, minPresValue, maxPresValue

**Reply by obscuresc** [reply]
I've edited my question to reflect more on what I was after, cheers though!

**Comment by bbartling** [comment]
You maybe to try bacpypes snapshot.

https://github.com/JoelBender/bacpypes-snapshot

It's a little bit more involved getting setup where you should study how to
setup legacy bacpypes before bacpypes snapshot but it would give you
everything you need like poll every possible BACnet attribute in your app.

If your a discord user join this server, it's good for BACnet talk...
Anyone can join just introduce your self in the intro section. Feel free to
DM me on there I can help you get setup.

https://discord.com/invite/2vvgmbCDhC

Am curious to get more into what you are doing with embedded devices and
testing applications in a test bench application.

Cheers.
